src/Combinatorics/EnumerativeCombinatorics/combinations.jl

```diff
@@ -88,7 +88,14 @@ function number_of_combinations(n::IntegerUnion, k::IntegerUnion)
   return binomial(ZZ(n), ZZ(k))
 end
 
-Base.length(C::Combinations) = BigInt(number_of_combinations(C.n, C.k))
+function Base.length(C::Combinations)
+  try
+    return Int(number_of_combinations(Int(C.n), Int(C.k)))
+  catch e
+    print("Length is too large, use `number_of_combinations` instead\n")
+    rethrow(e)
```

Comment:
Hmm, should we perhaps check `e` to see if that's really due to a the length being too large, and in that case throw a new error? Like this (untested)
```suggestion
  catch e
    e isa InexactError && error("Length is too large, use `number_of_combinations` instead")
    rethrow(e)
```